src/Model/IndexInfo.php

```diff
@@ -70,6 +70,42 @@ public function __toString()
         return $this->getName();
     }
 
+    /**
```

Comment:
Looking at the methods introduced in this PR, I think it'd be more worthwhile to add "is" methods for the various index types. That would be both more maintainable _and_ easier to document.

 * `is2d()`
 * `is2dSphere()`
 * `isGeoHaystack()`
 * `isText()`

Consider the following shell session:

```
> db.foo.createIndex({title:"text",description:"text"})
{
	"createdCollectionAutomatically" : true,
	"numIndexesBefore" : 1,
	"numIndexesAfter" : 2,
	"ok" : 1
}
> db.foo.createIndex({pos1:"2d"})
{
	"createdCollectionAutomatically" : false,
	"numIndexesBefore" : 2,
	"numIndexesAfter" : 3,
	"ok" : 1
}
> db.foo.createIndex({pos2:"2dsphere"})
{
	"createdCollectionAutomatically" : false,
	"numIndexesBefore" : 3,
	"numIndexesAfter" : 4,
	"ok" : 1
}
> db.foo.createIndex({pos3:"geoHaystack",x:1},{bucketSize:1})
{
	"createdCollectionAutomatically" : false,
	"numIndexesBefore" : 4,
	"numIndexesAfter" : 5,
	"ok" : 1
}
> db.foo.getIndexSpecs()
[
	{
		"v" : 2,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "test.foo"
	},
	{
		"v" : 2,
		"key" : {
			"_fts" : "text",
			"_ftsx" : 1
		},
		"name" : "title_text_description_text",
		"ns" : "test.foo",
		"weights" : {
			"description" : 1,
			"title" : 1
		},
		"default_language" : "english",
		"language_override" : "language",
		"textIndexVersion" : 3
	},
	{
		"v" : 2,
		"key" : {
			"pos1" : "2d"
		},
		"name" : "pos1_2d",
		"ns" : "test.foo"
	},
	{
		"v" : 2,
		"key" : {
			"pos2" : "2dsphere"
		},
		"name" : "pos2_2dsphere",
		"ns" : "test.foo",
		"2dsphereIndexVersion" : 3
	},
	{
		"v" : 2,
		"key" : {
			"pos3" : "geoHaystack",
			"x" : 1
		},
		"name" : "pos3_geoHaystack_x_1",
		"ns" : "test.foo",
		"bucketSize" : 1
	}
]
>
```

A robust method to implement `is<type>()` for each of these four types might be:

```php
return array_search('typeName', $this->getKey(), true) !== false;
```

This relies on `getKey()` to cast the "key" field to an array, since it is technically a document and could be a stdClass, and then uses [`array_search()`](http://php.net/array_search) to check if any field (i.e. key) has the type we're looking for. Roughly equivalent to:

```php
foreach ($this->info['key'] as $type) {
    if ($type === 'typeName') {
        return true;
    }
}

return false;
```